tests/Aspire.Hosting.Azure.Tests/AzurePostgresExtensionsTests.cs

```diff
@@ -323,4 +323,52 @@ public async Task WithPasswordAuthenticationBeforeAfterRunAsContainer(bool befor
         Assert.Equal("Host=localhost;Port=12455;Username=user;Password=p@ssw0rd1;Database=db1", await db1.Resource.ConnectionStringExpression.GetValueAsync(CancellationToken.None));
         Assert.Equal("Host=localhost;Port=12455;Username=user;Password=p@ssw0rd1;Database=db2Name", await db2.Resource.ConnectionStringExpression.GetValueAsync(CancellationToken.None));
     }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void RunAsContainerAppliesAnnotationsCorrectly(bool before)
```

Comment:
What happens if I do this?

```csharp
var db = builder.AddAzurePostgresFlexibleServer("pg")
                .WithPasswordAuthentication()
                .RunAsContainer(c =>
                {
                    c.WithPgAdmin();
                })
                .AddDatabase("db")
                .WithAnnotation(new DummyAnnotation());
```

Based on what I'm seeing in this branch the DummyAnnotation will be lost by the time this gets down into the application executor.

Reply:
Good catch! I've fix this and added tests for it.